From 71fdc766becb0d7b4c5a004a9c85a3adcdbdfdc0 Mon Sep 17 00:00:00 2001
From: Arnon Kaufman <akaufman@habana.ai>
Date: Tue, 14 Sep 2021 17:57:49 +0300
Subject: [PATCH] [SW-0] uart: print 64bits address

In order to print a 64 bits hex value rather then 32 bits value.
By default the preboot's print function prints a 32 bits
hex values, when trying to use those function with a 64 bits value,
only the lower (lsb) 32 bits is printed out.

Change-Id: I07fdde5279c7af36f21c370ca51417a36fe7fce2
Signed-off-by: Arnon Kaufman <akaufman@habana.ai>
---

diff --git a/lib/common/uart.c b/lib/common/uart.c
index 44d78f5..9f59d49 100644
--- a/lib/common/uart.c
+++ b/lib/common/uart.c
@@ -219,6 +219,13 @@
 	preboot_puth(val, pad);
 }
 
+void preboot_puts_hex64(const char *log, unsigned long long val, bool pad)
+{
+	preboot_puts_raw(log);
+	preboot_puts_raw("0x");
+	preboot_puth_raw((val >> 32), pad);
+	preboot_puth((val & 0xffffffff), true);
+}
 void preboot_puts_char(const char *str, unsigned char val)
 {
 	preboot_puts_hex(str, val, false);
diff --git a/lib/common/uart.h b/lib/common/uart.h
index 41698ab..ea59b68 100644
--- a/lib/common/uart.h
+++ b/lib/common/uart.h
@@ -19,6 +19,7 @@
 void preboot_puth(unsigned int val, bool pad);
 void preboot_puts_hex_raw(const char *log, unsigned int val, bool pad);
 void preboot_puts_hex(const char *log, unsigned int val, bool pad);
+void preboot_puts_hex64(const char *log, unsigned long long val, bool pad);
 void preboot_puts_char(const char *str, unsigned char val);
 bool preboot_uart_tx_full(void);
 void preboot_uart_tx_flush(void);
